app/auth/routes.py

```diff
@@ -83,3 +83,20 @@ def reset_password(token):
         flash('Your password has been reset.')
         return redirect(url_for('auth.login'))
     return render_template('auth/reset_password.html', title='Reset Password', form=form)
+
+@bp.route('/reset_password_preferences', methods=['GET', 'POST'])
+def reset_password_preferences():
+    user = current_user
+    if not user:
+        return redirect(url_for('main.index'))
+    form = ResetPasswordFormPreferences()
+    if form.validate_on_submit():
+        if (user.check_password(form.oldPassword.data)):
+            user.set_password(form.password.data)
+            db.session.commit()
+            flash('Your password has been reset.')
+            return redirect(url_for('main.index'))
+        flash('Incorrect Password.')
+        return render_template('auth/reset_password_preferences.html', title='Reset Password', form=form)
```

Comment:
With the validation written in the form class, these two lines won't be needed.

Reply:
You can also get rid of the: `if (user.check_password(form.oldPassword.data)):` in the route since it'll be validated by the form once implemented in the form class.